### DISCUSSION TITLE
Download of Library.sensor.image.lcd.KPU

### DISCUSSION THREAD

**Discussion by awchen9** [micropython/micropython#13761] [Libraries & Drivers]
Using thorny for esp32 cam development, missing sensor. image. lcd KPU, how to install and download.Here is my code
```import gc
import os
import machine
import time
import uos
import sensor
import image
import lcd
import KPU as kpu


sensor.reset()
sensor.set_pixformat(sensor.RGB565)
sensor.set_framesize(sensor.QVGA)
sensor.run(1)


lcd.init()


model_path = "/sd/module/best.tflite"
task = kpu.load(model_path)


labels = ["happy", "neutral", "angry", "yawn", "close"]


while True:
    img = sensor.snapshot()
    img = img.resize(224, 224)
    img_array = img.to_bytes()


    fmap = kpu.forward(task, img_array)


    output = fmap[:]


    for idx, value in enumerate(output):
        label = labels[idx]
        print("{}: {:.2f}%".format(label, value * 100))


    gc.collect()


    time.sleep(1)

kpu.deinit(task)

```

**Comment by andrewleech** [comment]
Where did this example code come from? I don't know of kpu or image modules for esp32, they're not a standard part of micropython that I'm aware of.

**Reply by awchen9** [reply]
`import gc
import os
import machine
import time
import uos
import sensor
import image
import lcd
import tensorflow as tf

sensor.reset()
sensor.set_pixformat(sensor.RGB565)
sensor.set_framesize(sensor.QVGA)
sensor.run(1)

lcd.init()

model_path = "/sd/module/best.tflite"
interpreter = tf.lite.Interpreter(model_path=model_path)
interpreter.allocate_tensors()

input_details = interpreter.get_input_details()
output_details = interpreter.get_output_details()

labels = ["happy", "neutral", "angry", "yawn", "close"]

while True:
    img = sensor.snapshot()
    img = img.resize(224, 224)

    img_array = img.to_bytes()

    interpreter.set_tensor(input_details[0]['index'], img_array)

    interpreter.invoke()

    output = interpreter.get_tensor(output_details[0]['index'])

    for idx, value in enumerate(output[0]):
        label = labels[idx]
        print("{}: {:.2f}%".format(label, value * 100))

    time.sleep(1)
`
This is the latest code
